### PR TITLE
add github actions CI script that builds on PR and pushes to master

### PR DESCRIPTION
**Pull request by flothesof**
Here’s a PR adding CI using Github actions.

It works on my fork of sfepy, see results here:
https://github.com/flothesof/sfepy/actions

This script builds and tests sfepy using pytest under windows, linux and macos.
It is configured to trigger for pushes to master and also pull requests to master.

What do you think about this @rc ?

### CODE REVIEW

**Comment by rc** [comment]
Thank you for taking a stab at this! I guess I need to enable something in the repo? :)

Maybe add a new section to the developer guide, explaining shortly the github setup for future reference.

IMO testing the PRs is enough - pushes to master do not need to be tested.

**Comment by flothesof** [comment]
Actions are enabled by default, I think. 
I’ll add something to the developer guide on this branch. 
And disable the build for pushes to master.

**Comment by flothesof** [comment]
@rc I added some documentation about the CI process.
I included it in the contribution guide, as I don’t think there’s lots to be said about it right now.

File: .github/workflows/build_and_test_matrix.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      run: |
+         pip install --upgrade pip
+         pip install flake8 pytest coverage wheel
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        # flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        flake8 . --count --select=E9,F63,F7 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```
Comment (rc) [inline .github/workflows/build_and_test_matrix.yml:33]: In this, we follow the NumPy/SciPy style guide which says 79 characters. The only exception can be when providing usage examples - then the command line can be as long as needed. So maybe disable this second test?

File: .github/workflows/build_and_test_matrix.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+         pip install --upgrade pip
+         pip install flake8 pytest coverage wheel
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        # flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        flake8 . --count --select=E9,F63,F7 --show-source --statistics
```
Comment (rc) [inline .github/workflows/build_and_test_matrix.yml:31]: We could use --select=E9,F63,F7,F82 after fixing the undefined variables that are just remnants of the Python2/3 compatibility code.

File: pyproject.toml [hunk 3]
```diff
@@ -1,13 +1,12 @@
 [build-system]
 # Minimum requirements for the build system to execute.
 requires = [
-    "setuptools<60.0",
+    "setuptools>=40.8.0",
```
Comment (rc) [inline pyproject.toml:4]: Use please `setuptools>=46.4,<60.0` - the upper bound is required for numpy.distutils to work.

**Comment by rc** [comment]
Thanks for the update! I have just a few questions - see above.

**Comment by flothesof** [comment]
That’s strange because I did something similar in my repo.
Maybe have a look at your settings? Mine look like this:
image

**Comment by rc** [comment]
I have the same settings as in your screenshot. Maybe I would need to merge and then make a test PR? When I open the actions tab, I get
Screenshot_20230208_112624

**Comment by flothesof** [comment]
My page shows detected actions, unlike yours (see https://github.com/flothesof/sfepy/actions). 

At this stage, I would recommend you try it out by merging, since it worked on my side... :)

File: .github/workflows/build_and_test_matrix.yml (hunk 2 above)
Comment (flothesof) [reply .github/workflows/build_and_test_matrix.yml:31]: Sure, I’ll let you do it :)

File: pyproject.toml (hunk 3 above)
Comment (flothesof) [reply pyproject.toml:4]: Changed in next commit.

File: pyproject.toml [hunk 4]
```diff
@@ -1,13 +1,12 @@
...
     "wheel",
-    "Cython>=0.29.30,<3.0",
-
+    "numpy",
+	"cython>=0.29.30",
```
[the remark this answers is not in the document]
Comment (flothesof) [reply pyproject.toml]: Changed in next commit.

**Pull request merged**